subprojects/core/src/main/java/org/gradle/internal/fingerprint/impl/DefaultFileCollectionSnapshotter.java

```diff
@@ -78,7 +77,6 @@ public void visitFileTree(File root, PatternSet patterns, FileTreeInternal fileT
                 root.getAbsolutePath(),
                 new PatternSetSnapshottingFilter(patterns, stat)
             )
-                .filter(snapshot -> snapshot.getType() != FileType.Missing)
```

Comment:
We've spent quite a bit of effort to remove this one line over a pretty long time. :) It's good to see it finally gone.